Sources/SwiftKuery/Table.swift

```diff
@@ -338,52 +304,37 @@ open class Table: Buildable {
      }
      var personTable = PersonTable()
      var employeeTable = EmployeeTable()
+     employeeTable = employeeTable.primaryKey([employeeTable.firstname, employeeTable.surname])
      personTable = personTable.foreignKey([personTable.firstName, personTable.lastName], references: [employeeTable.firstName, employeeTable.surname])
      ```
-    
+
      - Parameter columns: An Array of columns that constitute the foreign key.
      - Parameter references: An Array of columns from the foreign table that are referenced by the foreign key.
      - Returns: A new instance of `Table`.
-    */
+     */
     public func foreignKey(_ columns: [Column], references: [Column]) -> Self {
-        if foreignKeyColumns != nil || foreignKeyReferences != nil {
-            syntaxError += "Conflicting definitions of foreign key. "
-        }
-        else if columns.count == 0 || references.count == 0 || columns.count != references.count {
+        guard let newKey = ForeignKey(keys: columns, refs: references) else {
             syntaxError += "Invalid definition of foreign key. "
+            return self
         }
-        else if !Table.columnsBelongTo(table: self, columns: columns) {
-            syntaxError += "Foreign key contains columns from another table. "
-        }
-        else if !Table.columnsBelongTo(table: references[0].table, columns: references) {
-            syntaxError += "Foreign key references columns from more than one table. "
-        }
-        else {
-            foreignKeyColumns = columns
-            foreignKeyReferences = references
+        if !foreignKeys.contains(newKey) {
+            foreignKeys.append(newKey)
+        } else {
+            print("Duplicate foreign key detected")
```

Comment:
I assume this is a print because we still want to continue if we detect duplicate foreign keys. Do we want this to be a log instead of a print through?

Reply:
This was debug I had missed during my tidy up, will remove it.